File: segundo/1er cuatri/PDOO/ejercicios_clase/clase1811-ruby/mago.rb
# frozen_string_literal: true
require_relative 'jugador'

class Mago < Jugador
  def initialize(nombre, vida, nivel, hechizo)
    super(nombre, vida, nivel)
    @hechizo = hechizo
  end

  def aprender(nuevo_hechizo)
    @hechizo = nuevo_hechizo
  end

  def atacar
    super
    puts "con el hecho #{@hechizo} "
  end
end

File: segundo/1er cuatri/PDOO/ejercicios_clase/clase18-11/src/clase18/pkg11/Mago.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package clase18.pkg11;

/**
 *
 * @author alvarogz
 */
public class Mago extends Jugador{
    private String hechizo;
    
    public Mago(String nombre, int vida, int nivel, String hechizo){
        super(nombre, vida, nivel);
        this.hechizo = hechizo;
    }
    
    public void aprender(String nuevo_hechizo){
        hechizo = nuevo_hechizo;
    }
    public void atacar() {
        super.atacar();
        System.out.println(" con el hechizo " + this.hechizo);
    }
}
